summariser/src/analyze.rs

```diff
@@ -0,0 +1,301 @@
+use crate::model::{
+    PartitionKey, PartitionRateStats, PartitionTimingStats, PartitionedRateStats,
+    PartitionedTimingStats, StandardRateStats, StandardTimingsStats, TimingTrend,
+};
+use anyhow::Context;
+use itertools::Itertools;
+use polars::frame::DataFrame;
+use polars::prelude::*;
+
+pub(crate) fn standard_timing_stats(
```

Comment:
These are legible enough to not require tests for them. For more complicated stats it would be good to have tests for the formulas.

Reply:
Hmm, yeah we technically are testing them in the functional tests but you're right that it would be good to pass some data frames where the statistics it should calculate are easy to check against the actual results. That would be valid for these, they aren't that simple! :) I'm tempted to say though, let's see if this is any use before investing in unit tests for them on top of the functional tests

Reply:
Agreed 👍